configuration/configuration_test.go

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package configuration
+
+import (
+	"github.com/01org/ciao/payloads"
+	"io/ioutil"
+	"syscall"
+	"testing"
+)
+
+const badScheme = "badScheme://non-existent/path/nofile"
+const invalidURI = "file://%z invalid uri with spaces"
+const emptyPathURI = "file://"
+
+const keystoneURL = "http://keystone.example.com"
+const glanceURL = "http://glance.example.com"
+const computeNet = "192.168.1.110"
+const mgmtNet = "192.168.1.111"
+const storageURI = "/etc/ciao/configuration.yaml"
+const identityUser = "controller"
+const identityPassword = "ciao"
+const httpsKey = "/etc/pki/ciao/compute_key.pem"
+const httpsCACert = "/etc/pki/ciao/compute_ca.pem"
+
+const minValidConf = `---
+configure:
+    scheduler:
+        storage_uri: /etc/ciao/configuration.yaml
+    controller:
+        compute_ca: /etc/pki/ciao/compute_ca.pem
+        compute_cert: /etc/pki/ciao/compute_key.pem
+        identity_user: controller
+        identity_password: ciao
+    launcher:
+        compute_net: 192.168.1.110
+        mgmt_net: 192.168.1.111
+    image_service:
+        url: http://glance.example.com
+    identity_service:
+        url: http://keystone.example.com
+`
+
+func TestGetPayload(t *testing.T) {
+	var conf payloads.Configure
+
+	fillDefaults(&conf)
+	conf.Configure.Scheduler.ConfigStorageURI = storageURI
+	conf.Configure.Controller.HTTPSCACert = httpsCACert
+	conf.Configure.Controller.HTTPSKey = httpsKey
+	conf.Configure.Controller.IdentityPassword = identityPassword
+	conf.Configure.Launcher.ComputeNetwork = computeNet
+	conf.Configure.Launcher.ManagementNetwork = mgmtNet
+	conf.Configure.ImageService.URL = glanceURL
+	conf.Configure.IdentityService.URL = keystoneURL
+
+	payload, _ := getPayload(&conf)
+	if payload != nil {
+		t.Fatalf("expected nil, got [%s]", string(payload))
+	}
+
+	conf.Configure.Controller.IdentityUser = identityUser
+
+	_, err := getPayload(&conf)
+	if err != nil {
+		t.Fatalf("expected nil, got %v", err)
+	}
+
+}
+
+func TestConfigurationPayload(t *testing.T) {
+	var conf payloads.Configure
+	err := ConfigurationPayload(nil, &conf)
+	if err != nil {
+		t.Fatalf("Expected nil, got %v", err)
```

Comment:
`ConfigurationPayload()` on a nil blob should return an error, so I wonder if this is correct.